code/modules/food/food/condiment.dm

```diff
@@ -441,30 +453,87 @@
 
 //End of MRE stuff.
 
-/obj/item/weapon/reagent_containers/food/condiment/flour
-	name = "flour sack"
-	desc = "A big bag of flour. Good for baking!"
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour
+	name = "flour carton"
+	desc = "A big carton of flour. Good for baking!"
 	icon = 'icons/obj/food.dmi'
 	icon_state = "flour"
 	volume = 220
 	center_of_mass = list("x"=16, "y"=8)
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/on_reagent_change()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/on_reagent_change()
+	update_icon()
 	return
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/Initialize()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/Initialize()
 	. = ..()
 	reagents.add_reagent("flour", 200)
 	randpixel_xy()
 
+/obj/item/weapon/reagent_containers/food/condiment/advanced/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/food.dmi', src, "[icon_state]10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(-1 to 9)		filling.icon_state = "[icon_state]-10"
+			if(10 to 24) 	filling.icon_state = "[icon_state]-25"
+			if(25 to 49)	filling.icon_state = "[icon_state]-50"
+			if(50 to 85)	filling.icon_state = "[icon_state]-75"
+			if(86 to INFINITY)	filling.icon_state = "[icon_state]-100"
+
+		overlays += filling
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic
+	name = "flour sack"
+	desc = "An artisanal sack of flour. Classy!"
+	icon_state = "flour_bag"
```

Comment:
(Post sprite-review) Why are you applying the filled sprites as overlays to the empty sprite here? For the cartons, it makes sense how you're using it. But... the filled sprites are complete resprites. Why not just set the icon_state directly?

Reply:
I figured this would be simpler, letting them all inherit the same behavior, but I can add an if statement.